tutorials/integration.md

```diff
@@ -345,122 +388,134 @@ for more information see here](https://medium.com/jeremy-keeshin/hello-world-for
-communicate with the Ethereum network.
 ```javascript
-const ethers = require("ethers")
+var provider
+var userAddr   // The user's address
 ```
 
-Override these configuration parameters.
-```javascript
-const gsnConfig = configureGSN({
-	relayHubAddress: conf.relayhub,
-	paymasterAddress: conf.paymaster,
-	stakeManagerAddress: conf.stakemgr,
-	gasPriceFactorPercent: 70,
-	methodSuffix: '_v4',
-	jsonStringifyRequest: true,
-```
+&nbsp;
+
+This function is asynchronous so it could be called mutiple times.
+We use the fact that there's a value in `provider` to signal that
+access through GSN is available
 
-This is the `chainId` for the [Kovan test network](https://kovan-testnet.github.io/website/). 
-You can see the `chainId` values for other networks [here](https://chainid.network/).
 ```javascript
-	chainId: 42,
-	relayLookupWindowBlocks: 1e5
-})    // gsnConfig
+const startGsn = async () => {
+	await window.ethereum.enable()
+
+	if (provider != undefined)
+		return;
 ```
 
-The standard is to have the wallet manager in the browser (for example, MetaMask) expose 
-a [`Web3`](https://web3js.readthedocs.io/en/v1.2.0/web3.html) compatible provider in 
-`window.ethereum`. This provider is then wrapped by GSN, which creates a `RelayProvider` that 
-is also compatible with `Web3`. Transactions are processed by GSN to allow them to go through 
-a relay at zero cost, and then sent to the original wallet manager to be signed by the user.
+&nsbp;
+The standard is to have the wallet manager in the browser (for example, MetaMask) 
+expose a [`Web3`](https://web3js.readthedocs.io/en/v1.2.0/web3.html) 
+compatible provider in `window.ethereum`. This provider is then wrapped by 
+GSN, which creates a `RelayProvider` that is also compatible with `Web3`. 
+Transactions are processed by GSN to allow them to go through a relay at 
+zero cost, but they are sent to the original wallet manager to be 
+signed by the user.
+
+The `ethers.js` package uses its own provider class. So to create 
+the provider we’ll use, we take that GSN provider and use it as the 
+parameter to the `ethers` provider constructor.
 
-The `ethers.js` package uses its own provider class. So to create the provider we’ll use, 
-we take that GSN provider and use it as the parameter to the `ethers` provider constructor.
 ```javascript
-const origProvider = window.ethereum;
-const gsnProvider = new RelayProvider(origProvider, gsnConfig);
-const provider = new ethers.providers.Web3Provider(gsnProvider);
+
+        let gsnProvider = await new gsn.RelayProvider(window.ethereum, {
+		forwarderAddress: conf.forwarder,
+                paymasterAddress: conf.paymaster,
+                verbose: false}).init()
+	provider = new ethers.providers.Web3Provider(gsnProvider)
+	userAddr = gsnProvider.origProvider.selectedAddress
 ```
 
-Information about the contract we are accessing.
+&nbsp;
+
+For debugging purposes, it is useful to expose some variables in 
```

Comment:
@drortirosh 
Is this a general-purpose advice we are willing to give at this stage of a tutorial? It does not seem GSN-specific to me.